### PR TITLE
add CI hookup for dcos-checks repo

### PR DESCRIPTION
**Pull request by initcore0**
- add Dockerfile for reliable dcos-checks build
- add Jenkinsfile to orchestrate CI builds / tests
- update test.sh script to utilize docker environment for builds / tests
- fix some go fmt warnings

### CODE REVIEW

File: script/test.sh [hunk 1]
```diff
@@ -9,62 +9,114 @@
...
+    done
+
+    go tool cover -func ${BUILD_DIR}/coverage-reports/profile.cov
+    gocov convert ${BUILD_DIR}/coverage-reports/profile.cov \
+        | gocov-xml > "${BUILD_DIR}/coverage-reports/coverage.xml"
 }
 
 
-# Main.
+# Main. Example usage: ./test.sh collector unit
```
Comment (branden) [inline script/test.sh]: Is this example right? Looks like this script only takes one arg, `TEST_SUITE`.

File: script/test.sh [hunk 2]
```diff
@@ -9,62 +9,114 @@
...
+# The reports are saved to build/component/{test-reports,coverage-reports}/*.xml
 #
 set -e
 set -o pipefail
 export PATH="${GOPATH}/bin:${PATH}"
 
-PACKAGES="$(go list ./... | grep -v /vendor/)"
-SUBDIRS="cmd"
+if [ $# -ne 1 ]; then
+    echo "usage: $0 [test suite]"
```
Comment (branden) [inline script/test.sh]: `[]` usually indicates optional arguments, but the test suite appears to be a required argument. And `test suite` should be one word to indicate that it's a single arg.

How about:
```
echo "usage: $0 test_suite"
```

**Review by branden — COMMENTED** [review]
LGTM, just have some comments about usage docs in the test script.

File: script/test.sh [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-# This script performs tests against the dcos-metrics project, specifically:
+# This script performs tests against the dcos-backup project, specifically:
```
Comment (amitaekbote) [inline script/test.sh]: checks?

File: script/test.sh (hunk 1 above)
Comment (initcore0) [reply script/test.sh]: it was a copy paste from dcos-backup

**Review by amitaekbote — APPROVED** [review]

**Pull request merged**